### PR TITLE
Fix header icon

### PR DESCRIPTION
**Pull request by ashwin-pc**
### Description

Fixes header icon change

Screenshot 2023-04-20 at 5 26 01 PM

### Issues Resolved

https://github.com/opensearch-project/OpenSearch-Dashboards/issues/3909

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Review by ananzh — APPROVED** [review]

File: src/core/public/chrome/ui/header/header_help_menu.tsx [hunk 1]
```diff
@@ -330,13 +330,12 @@ class HeaderHelpMenuUI extends Component<Props, State> {
...
       >
-        <EuiIcon type="questionInCircle" size="m" />
         <EuiIcon
-          type="help"
+          type="questionInCircle"
           size="m"
           title={intl.formatMessage({
             id: 'core.ui.chrome.headerGlobalNav.helpMenuButtonTitle',
-            defaultMessage: 'Help',
+            defaultMessage: 'Help menu',
```
Comment (joshuarrrr) [inline src/core/public/chrome/ui/header/header_help_menu.tsx:338]: I think the finalized guidance is just "Help": https://github.com/opensearch-project/documentation-website/issues/3351#issuecomment-1466379362

**Review by kavilla — APPROVED** [review]

File: src/core/public/chrome/ui/header/header_help_menu.tsx (hunk 1 above)
Comment (joshuarrrr) [inline src/core/public/chrome/ui/header/header_help_menu.tsx:338]:
```suggestion
            defaultMessage: 'Help',
```

**Review by joshuarrrr — DISMISSED** [review]

**Review by kavilla — DISMISSED** [review]

**Comment by joshuarrrr** [comment]
pulled and verified

**Review by ananzh — APPROVED** [review]

**Review by joshuarrrr — APPROVED** [review]

**Comment by joshuarrrr** [comment]
verified unit tests and updated snapshots

**Review by kavilla — APPROVED** [review]

**Pull request merged**